### DISCUSSION TITLE
Uses of .graph !

### DISCUSSION THREAD

**Discussion by JosTheDude** [temper-studios/graph-file-format#3] [General]
What are you using the format for!

I'm considering using it for my MC Plugins as a sort of data structures for my player-script made variables, as utilizing a csv file / mysql becomes inefficient when the variable count reaches a certain threshold. Just wondering how much the file format can really take in terms of data, and how optimized it is :)

**Comment by temper-studios** [comment]
I have not tested performance nor done a round of optimisations. But, by it's design it's pretty fast by default because everything is done in-place (allocations are done all at once and not while it is being parsed). There is a potential optimisation that I haven't got round to doing yet that will make it faster aswell.

You'd have to try it out. I store game scenes with these files as I work with them. They are around 400kb and I haven't experienced much slow down, but mileage will vary.

As for performant files probably best to stick to binary ones.

**Reply by temper-studios** [reply]
Did a rudimentary test and I can load a 380kb file in about 0.007 seconds (7 ms) on my run of the mill laptop. So could be faster.
The slow part will likely be traversing the tree

**Reply by temper-studios** [reply]
I was wrong actually the file is 550kb

**Reply by JosTheDude** [reply]
Did a little test, the implementation is **very** makeshift and is probably no where close to how it can be implemented into my code in production, but attempting to store live updating songs corresponding to the player's location and settings appears to be working well. Granted, this is pulling stuff from our original system (the player location and song list), but it's a start :D. I'll try to experiment more and see what I can do, but I might make an API to implement this as an alternative to the norm of sqlite stored variables / csv stored variables in the mc community. I'm most likely going to attempt stress tests next though :)

I'll probably release the code for the public once it uh... doesn't look as hideous as it is right now :')

**Reply by temper-studios** [reply]
Can't tell if it's good or bad?

**Reply by JosTheDude** [reply]
Good... I think WKJHWLQEWQ

**Reply by temper-studios** [reply]
Dunno what you mean
